packages/select/src/components/omnibar/omnibar.tsx

```diff
@@ -94,9 +94,14 @@ export class Omnibar<T> extends React.PureComponent<IOmnibarProps<T>, IOmnibarSt
         const { isOpen } = nextProps;
         const canClearQuery = !this.props.isOpen && isOpen && this.props.resetOnSelect;
 
+        const query =
+            nextProps.inputProps && nextProps.inputProps.value
+                ? (nextProps.inputProps.value as string)
+                : this.state.query;
```

Comment:
while this is the right idea, it's not enough to actually solve this issue. you've fixed `Omnibar` but not `Select` or any of the others. i'm actually working on controlled support _right now_ so perhaps just hang tight?